godot_project/project_workspace/workspace_context/history/history.gd

```diff
@@ -80,8 +80,9 @@ var _workspace_context: WorkspaceContext
 var _snapshotable_systems: Array[Object]
 var _snapshot_stack: Array[Dictionary]
 var _name_stack: Array[String] = []
+var _version_stack: Array[int] = []
 var _stack_pointer: int = -1
-var _version: int
+var _highest_version: int = -1
```

Comment:
I don't like adding yet another array to track a parameter that should be attached to the snapshot directly but I'm not sure if we can add random data to a snapshot without affecting the rest.